linbox/ring/gcd32.h

```diff
@@ -1,195 +0,0 @@
-/* Copyright (C) LinBox
```

Comment:
I am not sure we want to remove gcd32.h and gcd64.h: they are used in `ring/poweroftwomodular.h` which has not been deleted in this PR.
Also it seems beyond a cleanup of Makefiles and would probably break users code based on poweroftwomodular.h